crates/net/network/src/transactions/mod.rs

```diff
@@ -1332,59 +1335,53 @@ impl TransactionSource {
 /// Tracks transactions a peer has seen.
 #[derive(Debug)]
 struct TransactionsSeenByPeer {
-    /// Keeps track of transactions that we know the peer has seen because they were announced by
-    /// the peer. It's possible that these transactions are pending fetch.
-    transactions_received_as_hash: LruCache<B256>,
-    /// Keeps track of transactions that we know the peer has seen because they were received in
-    /// full from the peer or sent to the peer.
-    transactions_received_in_full_or_sent: LruCache<B256>,
+    /// Keeps track of transactions that we know the peer has seen.
+    transactions: LruCache<B256>,
 }
 
 impl TransactionsSeenByPeer {
     /// Returns `true` if peer has seen transaction.
     fn has_seen_transaction(&self, hash: &TxHash) -> bool {
-        self.transactions_received_in_full_or_sent.contains(hash) ||
-            self.transactions_received_as_hash.contains(hash)
+        self.transactions.contains(hash)
     }
 
     /// Inserts a transaction hash that has been seen in an announcement.
     fn seen_in_announcement(&mut self, hash: TxHash) {
-        _ = self.transactions_received_as_hash.insert(hash);
+        // todo: add metrics
+        _ = self.transactions.insert(hash);
     }
 
     /// Inserts a hash of a transaction that has either been sent to the peer, or has been
     /// received in full from the peer over broadcast.
     fn seen_by_peer_and_in_pool(&mut self, hash: TxHash) {
-        _ = self.transactions_received_in_full_or_sent.insert(hash);
+        // todo: add metrics
+        _ = self.transactions.insert(hash);
     }
 
     /// Inserts a list of transactions that have either been sent to the peer, or have been
     /// received in full from the peer over broadcast.
     fn extend_seen_by_peer_and_in_pool(&mut self, hashes: impl IntoIterator<Item = TxHash>) {
-        self.transactions_received_in_full_or_sent.extend(hashes)
+        // todo: add metrics
+        self.transactions.extend(hashes)
```

Comment:
since this now doesn't distinguish between these cases, I'm thinking we should just combine these methods?

Reply:
sure, though I was wondering if we can move some metrics code in there since we have `messages_with_already_seen_hashes` and `messages_with_already_seen_transactions`, so left it for now. will just address it within this prs scope though since you bring it up.

Reply:
I concluded that it makes more sense to add the metrics I want to observe in another scope. removed this type and reverted `seen_transactions` to the way it was done before (just an `LruCache`).

clarified the meaning of the `already_seen_transactions`/`already_seen_hashes` metric as a result, which wasn't clear to me.

addressing this, I found a bug, and fixed it here https://github.com/paradigmxyz/reth/pull/6723.